### ISSUE TITLE
Heavy ads crypto currency mining

### ISSUE THREAD

**Issue by trimechee** [Eloston/ungoogled-chromium#1944]
Hello, in the privacy and security section, site settings, I noticed "that there is no native ad blocker integrated in Chromium, this ad blocker integrated in Chromium blocks heavy ads and malicious crypto currency mining sites , it implement steps to block ads that consume a large share of device resources before the user interacts with the ad. can you reactivate this very important option please ?

it is necessary and more efficient than adblock extensions which may not detect crypto mining currency:

Motivating Use Cases
Examples of observed ad behaviors that are intended to be discouraged:

Ads that mine cryptocurrency
Ads that load large, poorly compressed images
Ads that load large video files before a user gesture
Ads that perform expensive operations in javascript, such as decoding video files, or CPU timing attacks

The user agent will unload ads that use an egregious amount of network bandwidth or CPU usage.

https://developer.chrome.com/blog/heavy-ad-interventions/

https://github.com/chromium/chromium-ads-detection

https://github.com/johnivdel/heavy-ad-intervention

https://blog.chromium.org/2020/05/resource-heavy-ads-in-chrome.html

https://www.tomsguide.com/how-to/how-to-block-heavy-ads-in-chrome

**Comment by PF4Public** [comment]
Do I get it right: there is something in Chromium (not Chrome?) that blocks ads, right? Is it somehow disabled by ungoogled-chromium patches?

**Comment by trimechee** [comment]
Yes, i used chromium made by @Hibbiki then i moved again to Ungoodled-Chromium made by @Marmaduke, chromium version has this internal ad blocker, chromium has deleted also the falg reader mode but in Ungoogled-Chromium , this flag still exists but Marmaduke told me that it doesn't work anymore.....so it will be great to add this adblock just as an option and no nedd need to be activated by default, this adblock transmits data to google ? that's why Ungoogled-Chromium deactivated it?

**Comment by Ph0rk0z** [comment]
Is this like the ad blocker in bromite? If so, it's not so good and I'd rather have ublock origin, always.

**Comment by trimechee** [comment]
I dont"t knows, but chromium has a native mining blocker, ublock is good, also  there are new sites malicious advertising heavy and ublock cannot detect everything or we have to put a lot of lists which can have false positives make navigation heavier .... in addition crypto mining techniques are becoming so sophisticated that mining just uses a small amount of cpu power without user  noticing it, a press article talked about it ....

**Comment by trimechee** [comment]
I discover that flags that are responsible to block heavy ads and crypto mininig are still exits in Ungoogled- Chromium, it will be great to not delete them in order to enable these flags when needed may be , description says these flags are expired, in chromium these flags doesn't any more exist because they are enable by default : 


Heavy Ad Intervention
!!! NOTE: THIS FLAG IS EXPIRED AND MAY STOP FUNCTIONING OR BE REMOVED SOON !!! Unloads ads that use too many device resources. – Mac, Windows, Linux, Chrome OS, Android, Fuchsia

[#enable-heavy-ad-intervention](chrome://flags/#enable-heavy-ad-intervention)

Default
Heavy ad privacy mitigations
!!! NOTE: THIS FLAG IS EXPIRED AND MAY STOP FUNCTIONING OR BE REMOVED SOON !!! Enables privacy mitigations for the heavy ad intervention. Disabling this makes the intervention deterministic. Defaults to enabled. – Mac, Windows, Linux, Chrome OS, Android, Fuchsia

[#heavy-ad-privacy-mitigations](chrome://flags/#heavy-ad-privacy-mitigations)

**Comment by jstkdng** [comment]
bruh just block ads altogether with an extension, I'm sure ad providers have alredy found a way to mine without triggering the intervention.

**Comment by trimechee** [comment]
somes sites require us to disable the ad blocker to be able to access the site, they have an ad blocker detector, so the built-in native blocker is our lifeline because the native blocker allow the light ads but disable heavy ads

**Comment by PF4Public** [comment]
From the website perspective there is no big difference be it a "built-in native" or "extension" adblocker.

**Comment by trimechee** [comment]
I don't know, may be.....Ungoogled chromium in the download section, what is the name of the developer who have made the windows 64 bit version please ? https://ungoogled-software.github.io/ungoogled-chromium-binaries/

**Comment by networkException** [comment]
The name is contained in the release information for a specific release

**Comment by trimechee** [comment]
That's good, thank you

**Comment by Ahrotahn** [comment]
This feature is [enabled by default](https://source.chromium.org/chromium/chromium/src/+/refs/tags/101.0.4951.54:components/heavy_ad_intervention/heavy_ad_features.cc) and ungoogled-chromium doesn't explicitly disable it.  I haven't dug into this, but it's possible building without reporting or field trials causes this to not to function correctly.  

Google removed that entry from `chrome://flags` but the flag still exists.  We have a patch that allows those flags to continue to function until they are removed fully.  That is why you see the expiration note for them.  

There aren't any user-facing settings for this in vanilla chromium that I'm aware of.  Were you using someone's customized build to get the options you're looking for?  

At any rate I'll echo the recommendations to use Ublock Origin.

**Comment by trimechee** [comment]
@Ahrotahn Thank you very much fot these explications :) i used chromium made by Hibbiki in woolyss site which has heavy ads option in site parametrs, and recently i move again to Ungoogled chromium made by Marmaduke which hasn't this option in site parametres.......


https://chromium.woolyss.com/


but may be Ungoogled chroimum made by Nifury has this option like chromium, i don't know.......

https://ungoogled-software.github.io/ungoogled-chromium-binaries/releases/windows/64bit/101.0.4951.54-1

https://github.com/Nifury

**Comment by trimechee** [comment]
@Ahrotahn @PF4Public I test with with the last Ungoogled-Chromium found in Woolyss
website, i enable Heavy Ad Intervention flag and disable Heavy ad privacy mitigations, in this website which test heavy ads and heavy ads are not unloaded after ~30s of CPU usage if the tab is kept focused, so it's offical, Ungoodled-Chromium don't disable the heavy ads :(

https://heavy-ads.glitch.me/?ad=%2Fadserver-cpu%2Fadunit.html&n=1645302135593

**Comment by trimechee** [comment]
Now i test with microsoft edge, heavy ads are unloaded, this is the 2 report of cpu and network  : "[Reports from ads that trigger an intervention will show here]
{
  "type": "intervention",
  "url": "https://heavy-ads.glitch.me/adserver-cpu/adunit.html?n=1652124899185",
  "body": {
    "sourceFile": null,
    "lineNumber": null,
    "columnNumber": null,
    "id": "HeavyAdIntervention",
    "message": "Ad was removed because its peak CPU usage exceeded the limit. See https://www.chromestatus.com/feature/4800491902992384?utm_source=devtools"
  }
}




[Reports from ads that trigger an intervention will show here]
{
  "type": "intervention",
  "url": "https://heavy-ads.glitch.me/adserver-network/adunit.html?n=1652124953193",
  "body": {
    "sourceFile": null,
    "lineNumber": null,
    "columnNumber": null,
    "id": "HeavyAdIntervention",
    "message": "Ad was removed because its network usage exceeded the limit. See https://www.chromestatus.com/feature/4800491902992384?utm_source=devtools"
  }
}

**Comment by trimechee** [comment]
i found that there is third party filter list, so may be this is why Ungoogled Chromium has disable this option but i think this option is too important and shouldn't be disabled speceally fot those who visit streaming sites....
 https://github.com/chromium/chromium-ads-detection/tree/master/third_party/filter_lists
